sktm/jenkins.py

```diff
@@ -320,6 +326,56 @@ def build(self, baserepo=None, ref=None, baseconfig=None,
         logging.info("submitted build: %s", build)
         return build.get_number()
 
+    def __get_job(self):
+        """
+        Get Jenkens job by job name. Retry Jenkins N times in case of temporary
+        network failures.
+
+        Return:
+            (job, None) if succeed, else return (None, e) where e is the last
```

Comment:
Since the exception is going to be raised anyways, wouldn't this be too complicated? Can we just return the job if we succeed, and raise the exception if we can't reach Jenkins at all?